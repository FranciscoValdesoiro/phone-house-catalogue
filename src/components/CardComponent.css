* { 
    box-sizing: border-box;
}

a {
    text-decoration: none;
    color: var(--font-color)
}

.card-container:visited {
    color: var(--main-brand-color-visited);
}

.card-container {
    margin: 20px auto;
    max-width: 66%;
    min-width: 320px;
    color: var(--font-color);
    border: solid 5px white;
    background-color: white;
    text-align: center;
    font-size: larger;
    box-shadow: -2px 7px 12px 0 rgba(0,0,0,.12);
}

.card-container:hover, .card-container:active {
    border: solid 5px var(--main-brand-color);
    color: var(--main-brand-color);
}

.content {
    background-color: white;
}

.phone-image {
    max-width: 310px;
    max-height: 310px;
    min-height: 310px;
    
}

