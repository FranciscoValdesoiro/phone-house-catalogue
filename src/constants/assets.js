export const IMAGES_DIR = '/images/'