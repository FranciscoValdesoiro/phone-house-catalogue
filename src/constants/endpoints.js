export const GET_PHONES_URL = 'http://localhost:8080/phones';