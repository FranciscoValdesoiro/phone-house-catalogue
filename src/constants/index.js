export * from './endpoints';
export * from './assets';