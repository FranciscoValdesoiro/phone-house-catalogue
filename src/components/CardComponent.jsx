import React from 'react';
import './CardComponent.css'
import PropTypes from 'prop-types'

const CardComponent = ({name, imgSrc}) => {

  return (
    <div className="card-container">
      <h3>{name}</h3>
      <div className="content">
        <img className="phone-image" src={imgSrc} alt={name + 'phone image'}/>
      </div>
      <p>Click me for more details</p>
    </div>
  )           
}

CardComponent.propTypes = {
  name: PropTypes.string.isRequired,
  imgSrc: PropTypes.string.isRequired
}

export default CardComponent;
