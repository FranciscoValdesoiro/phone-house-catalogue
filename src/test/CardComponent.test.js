import React from 'react';
import { shallow } from 'enzyme'
import CardComponent from '../components/CardComponent';
import { IMAGES_DIR } from '../constants'
import '@testing-library/jest-dom'


describe('CardComponent', () => {

  it('Should render the CardComponent ', () => {
    const name = 'Iphone 7'
    const wrapper = shallow( <CardComponent name={name} imgSrc={IMAGES_DIR + 'iphone7.png'}/>)
    expect( wrapper ).toMatchSnapshot()
  })

});