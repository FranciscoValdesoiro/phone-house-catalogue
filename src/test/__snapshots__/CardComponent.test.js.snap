// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent Should render the CardComponent  1`] = `
<div
  className="card-container"
>
  <h3>
    Iphone 7
  </h3>
  <div
    className="content"
  >
    <img
      alt="Iphone 7phone image"
      className="phone-image"
      src="/images/iphone7.png"
    />
  </div>
  <p>
    Click me for more details
  </p>
</div>
`;
